src/app/backend/dashboard.go

```diff
@@ -71,6 +71,7 @@ var (
 	argSystemBannerSeverity      = pflag.String("system-banner-severity", "INFO", "Severity of system banner. Should be one of 'INFO|WARNING|ERROR'. Default: 'INFO'.")
 	argAPILogLevel               = pflag.String("api-log-level", "INFO", "Level of API request logging. Should be one of 'INFO|NONE|DEBUG'. Default: 'INFO'.")
 	argDisableSettingsAuthorizer = pflag.Bool("disable-settings-authorizer", false, "When enabled, Dashboard settings page will not require user to be logged in and authorized to access settings page.")
+	argNamespace                 = pflag.String("namespace", "kube-system", "When non-default namespace is used, create encryption key in the specified namespace. Default: 'kube-system'.")
```

Comment:
I suppose this should be the namespace that the dashboard is deployed in? Maybe we should allow setting this via an environment variable? Setting a value via an argument using the downward API is a bit wonky.

Maybe just make the default into os.GetEnv("POD_NAMESPACE") and then to 'kube-system' if that's empty?

```
var defNamespace = getEnv("POD_NAMESPACE", "kube-system")
var argNamespace  = pflag.String("namespace", defNamespace, "When non-default namespace is used, create encryption key in the specified namespace. Default: '"+defNamespace+"'.")

func getEnv(name, def string) string {
  val := os.GetEnv(name)
  if val == "" {
    val = def
  }
  return val
}
```

Reply:
I'm also curious why we don't already set POD_NAME, POD_NAMESPACE, POD_IP in the src/deploy/**/*.yaml files since it seems to be used in src/app/backend/cert/ecdsa/creator.go?

That might be a question for @maciaszczykm though.

Reply:
Yes please to leveraging environment variable instead of (or at least in addition to) a new flag.
Then YAMLs can simply add:

```yaml
      env:
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
```